tests/manage/pv_services/test_rwo_pvc_fencing_unfencing.py

```diff
@@ -0,0 +1,632 @@
+                select other 2 nodes where mon/osd are not running
+            Dedicated scenario: 3 Non-OCS nodes
+        - Disrupt the leader provisioner pods if not running on above selected
+            nodes
+        - Make the nodes (where app pods are running) unresponsive
+            by bringing their main network interface down
+        - Check new app pods and/or mon, osd pods scheduled on another node and
+            are stuck due to Multi-Attach error.
+        - Power off the unresponsive node
+        - Force delete the app pods and/or mon, osd pods on the unresponsive node
+        - Check new app pods and/or mon, osd pods scheduled on another node comes
+            into Running state
+        - Run IOs on new app pods
+        """
+        project, dc_pods, ceph_pods, app_pod_nodes, disruptor = setup
+
+        # Do setup and run IO on pods
+        md5sum_data = self.setup_pods_run_ios(
+            pod_list=dc_pods, fio_filename='io_file1'
+        )
+
+        # Run more ios in Background PR 1369
+        # pod.run_io_in_bg(dc_pods, expect_to_fail=True, fedora_dc=True)
+
+        # OCS-1430/OCS-1435
+        # Disrupt leader plugin-provisioner pods, skip if running on node to be failed
+        if disruptor:
+            [disruption.delete_resource() for disruption in disruptor]
+
+        # Induce network failure on the nodes
+        self.node_network_failure(app_pod_nodes)
+
+        # Wait for pods to be rescheduled
+        for pod_obj in (dc_pods + ceph_pods):
+            pod_obj.ocp.wait_for_resource(
+                condition=constants.STATUS_TERMINATING,
+                resource_name=pod_obj.name, timeout=420, sleep=60
+            )
+
+        # Fetch info of new pods and verify Multi-Attach error
+        new_dc_pods = self.get_new_pods(dc_pods)
+        assert len(new_dc_pods) == len(dc_pods), 'Unexpected number of app pods'
+        self.verify_multi_attach_error(new_dc_pods)
+
+        if ceph_pods:
+            new_ceph_pods = self.get_new_pods(ceph_pods)
+            assert len(new_ceph_pods) == len(ceph_pods), 'Unexpected number of mon/osd pods'
+            self.verify_multi_attach_error(new_ceph_pods)
+
+        # Power off the unresponsive node
+        nodes.stop_nodes(app_pod_nodes)
+
+        # Force delete the app pods and/or mon,osd pods on the unresponsive node
+        for pod_obj in (dc_pods + ceph_pods):
+            pod_obj.delete(force=True)
+
+        # Wait for new app pods to come to Running state
+        for pod_obj in new_dc_pods:
+            pod_obj.ocp.wait_for_resource(
+                condition=constants.STATUS_RUNNING, resource_name=pod_obj.name,
+                timeout=600, sleep=30
+            )
+
+        # Wait for mon and osd pods to come to Running state
+        pod_obj = ocp.OCP(
+            kind=constants.POD, namespace=config.ENV_DATA['cluster_namespace']
+        )
+        assert pod_obj.wait_for_resource(
```

Comment:
please add assert messages. Apply to all relevant places you assert

Reply:
Done